metadata-io/src/main/java/com/linkedin/metadata/system_info/collectors/SpringComponentsCollector.java

```diff
@@ -0,0 +1,266 @@
+   * endpoint calls this method - This method makes HTTP call to /system-info (same endpoint!) -
+   * Results in "request timed out" errors after 5 seconds
+   *
+   * <p>PLACEHOLDER SOLUTION: Instead of making HTTP calls that cause circular dependency, we return
+   * placeholder data that maintains the API structure while avoiding the issue.
+   *
+   * <p>FUTURE IMPROVEMENT: Detect deployment mode and handle accordingly: - Embedded: Return actual
+   * component status without HTTP calls (check internal state) - Separate: Make HTTP calls to
+   * external consumer service endpoints
+   */
+  private ComponentInfo getMceConsumerInfo() {
+    // Return placeholder to avoid circular dependency in embedded mode
+    // In embedded mode, MCE consumer runs in the same process as this code
+    return ComponentInfo.builder()
+        .name(MCE_COMPONENT_NAME)
+        .status(ComponentStatus.AVAILABLE) // Assume available since we're running
+        .version(gitVersion.getVersion()) // Same version as GMS in embedded mode
+        .properties(
+            Map.of(
+                "mode", "placeholder",
+                "deployment", "embedded",
+                "note",
+                    "Consumer info collection temporarily disabled to avoid circular dependency"))
+        .build();
+  }
+
+  /**
+   * Fetches component information from a remote service via HTTP.
+   *
+   * <p>NOTE: This method is currently not used to avoid circular dependency issues in embedded
+   * mode. See getMaeConsumerInfo() and getMceConsumerInfo() for details.
+   *
+   * <p>This method would be used in a future implementation that properly detects when consumers
+   * are running in separate containers and can safely make HTTP calls.
+   *
+   * <p>SECURITY NOTE: The baseUrl parameter comes from system configuration
+   * (maeConsumerUrl/mceConsumerUrl) which is set by administrators via environment variables or
+   * config files. This is not user input and does not represent an SSRF vulnerability. The URLs are
+   * controlled by the same administrators who manage database connections and other infrastructure
+   * settings.
+   */
+  // semgrep:ignore java.lang.security.audit.ssrf.ssrf-injection
+  // nosec: SSRF - URL comes from admin-controlled system configuration
+  // aikido:ignore
+  @SuppressWarnings({"security:ssrf", "aikido:ssrf"})
+  private ComponentInfo fetchRemoteComponentInfo(
+      String name, String baseUrl, String componentField) {
+    try {
+      HttpClient client = HttpClient.newBuilder().connectTimeout(Duration.ofSeconds(5)).build();
+
+      HttpRequest request =
+          HttpRequest.newBuilder()
+              .uri(URI.create(baseUrl + SYSTEM_INFO_ENDPOINT))
+              .timeout(Duration.ofSeconds(5))
+              .GET()
+              .build();
+
+      HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
```

Comment:
this is the copy of the fetchRemoteComponentInfo from the original PR, and I'm curious why you didn't use the async version of the method, @david-leifker, I think this way it would've been a little more lightweight and I think we wouldn't need the thread pool